metals/src/main/scala/scala/meta/internal/metals/JavaFormattingProvider.scala

```diff
@@ -0,0 +1,230 @@
+    }
+  }
+
+  private lazy val defaultSettings =
+    DefaultCodeFormatterOptions.getEclipseDefaultSettings.getMap.asScala.toMap
+
+  private def loadEclipseFormatConfig: Map[String, String] = {
+    userConfig().eclipseFormatConfigPath
+      .map(eclipseFormatFile => {
+        if (eclipseFormatFile.exists) {
+          val text = eclipseFormatFile.toInputFromBuffers(buffers).text
+          parseEclipseFormatFile(text) match {
+            case Failure(e) =>
+              scribe.error(
+                s"Failed to parse $eclipseFormatFile. Using default formatting",
+                e
+              )
+              defaultSettings
+            case Success(values) =>
+              values
+          }
+        } else {
+          scribe.warn(
+            s"$eclipseFormatFile not found.  Using default java formatting"
+          )
+          defaultSettings
+        }
+      })
+      .getOrElse(defaultSettings)
+  }
+
+  def format(
+      params: l.DocumentFormattingParams
+  ): Future[util.List[l.TextEdit]] = {
+    Future {
+      val options = params.getOptions
+      val path = params.getTextDocument.getUri.toAbsolutePath
+      val input = path.toInputFromBuffers(buffers)
+      runFormat(path, input, options, fromLSP(input)).asJava
+    }
+  }
+
+  private def fromLSP(input: Input, range: l.Range): m.Position.Range =
+    m.Position.Range(
+      input,
+      range.getStart().getLine(),
+      range.getStart().getCharacter(),
+      range.getEnd().getLine(),
+      range.getEnd().getCharacter()
+    )
+
+  private def fromLSP(input: Input): m.Position.Range =
+    m.Position.Range(input, 0, input.chars.length)
+
+  def format(
+      params: l.DocumentRangeFormattingParams
+  ): util.List[l.TextEdit] = {
+    val options = params.getOptions
+    val range = params.getRange
+    val path = params.getTextDocument.getUri.toAbsolutePath
+    val input = path.toInputFromBuffers(buffers)
+    runFormat(path, input, options, fromLSP(input, range)).asJava
+  }
+
+  def format(
+      params: l.DocumentOnTypeFormattingParams
+  ): util.List[l.TextEdit] = {
+    Nil.asJava
+  }
+
+  private def runFormat(
+      path: AbsolutePath,
+      input: Input,
+      formattingOptions: l.FormattingOptions,
+      range: m.Position.Range
+  ): List[l.TextEdit] = {
+    // if source/target/compliance versions aren't defined by the user then fallback on the build target info
+    var options = loadEclipseFormatConfig
+    if (
+      !options.contains(JavaCore.COMPILER_SOURCE) ||
+      !options.contains(JavaCore.COMPILER_CODEGEN_TARGET_PLATFORM) ||
+      !options.contains(JavaCore.COMPILER_COMPLIANCE)
+    ) {
+
+      val version = for {
+        targetID <- buildTargets.sourceBuildTargets(path)
+        java <- buildTargets.javaTarget(targetID)
+        sourceVersion <- java.sourceVersion
+        targetVersion <- java.targetVersion
+      } yield ((sourceVersion, targetVersion))
+
+      version
+        .take(1)
```

Comment:
```suggestion
        .headOption
```

So that it doesn't break if the target is not available?